### PR TITLE
Cache grouping optimization

### PR DESCRIPTION
**Pull request by z5830865**
Cache grouping optimization

### CODE REVIEW

**Review by z5830865 — COMMENTED** [review]
Cache grouping optimization

**Comment by gqcn** [comment]
@z5830865 Would you please give some description and business scenario about the `Group` feature?

**Comment by z5830865** [comment]
> @z5830865 Would you please give some description and business scenario about the `Group` feature?

Cache grouping allows you to group the associated caches together, similar to the clear function. You can group all caches of a table into a group. Reading the source code, clear does not implement such a feature

File: database/gdb/gdb_model_cache.go [hunk 1]
```diff
@@ -140,6 +172,16 @@ func (m *Model) saveSelectResultToCache(
 	if errCache := cacheObj.Set(ctx, cacheKey, cacheItem, m.cacheOption.Duration); errCache != nil {
 		intlog.Errorf(ctx, `%+v`, errCache)
 	}
+
+	if m.cacheEnabled && len(m.cacheOption.Group) > 0 {
+		keyGroup, errCache := cacheObj.Get(ctx, m.cacheOption.Group)
+		if errCache == nil {
+			keys := append(keyGroup.Strings(), cacheKey)
+			if errCache = cacheObj.Set(ctx, m.cacheOption.Group, keys, m.cacheOption.Duration); errCache != nil {
```
Comment (gqcn) [inline database/gdb/gdb_model_cache.go:180]: The `Get` and `Set` function is not atomic, which will cause concurrent safety issue.

**Review by gqcn — CHANGES_REQUESTED** [review]

File: database/gdb/gdb_model_cache.go (hunk 1 above)
Comment (z5830865) [reply database/gdb/gdb_model_cache.go:180]: You don't have to worry about concurrency security, and there's no problem with cache updates overwriting the original value

Comment (gqcn) [reply database/gdb/gdb_model_cache.go:180]: @z5830865 Would it loose keys when there were multiple goroutines doing `Get` and `Set`?

**Review by gqcn — CHANGES_REQUESTED** [review]

File: database/gdb/gdb_model_cache.go (hunk 1 above)
Comment (gqcn) [reply database/gdb/gdb_model_cache.go:180]: It is not good design using `Get/Set` to maintain the keys, as there's concurrent issue here.

**Review by gqcn — CHANGES_REQUESTED** [review]

**Pull request closed without merging**